### PR TITLE
Add manifest file generator and RequestCollector

### PR DESCRIPTION
**Pull request by rustamwin**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

**Comment by xepozz** [comment]
Would be nice to separate `WebAppInfoCollector` and `RequestCollector`

File: src/Collector/RequestCollector.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            $this->requestMethod = $event->getRequest()->getMethod();
+            $this->requestIsAjax = $event->getRequest()->getHeaderLine('X-Requested-With') === 'XMLHttpRequest';
+            $this->userIp = $event->getRequest()->getServerParams()['REMOTE_ADDR'] ?? null;
+        }
+        if ($event instanceof AfterRequest) {
+            $this->responseStatusCode = $event->getResponse()->getStatusCode();
+        }
+    }
+
+    private function reset(): void
```
Comment (xepozz) [inline src/Collector/RequestCollector.php:42]: Is it used anywhere?

Comment (rustamwin) [reply src/Collector/RequestCollector.php:42]: Yes https://github.com/yiisoft/yii-debug/blob/d2c6faa98dbd62c3baf83f2ade8f0eb9137379bd/src/Collector/CollectorTrait.php#L16

**Comment by xepozz** [comment]
I don't like this solution:
- I don't understand for what purpose the manifest is needed
- FileStorage now depends on concrete collectors

**Comment by rustamwin** [comment]
Decided to add only the request collector, see #48 

If we need a manifest later, we'll add it.

**Pull request closed without merging**